src/widgets/table/table.rs

```diff
@@ -727,30 +725,23 @@ impl Table<'_> {
                 .map(|r| r.cells.len())
                 .max()
                 .unwrap_or(0);
-            // There are `col_count - 1` spaces between the columns
-            let total_space =
-                max_width.saturating_sub(self.column_spacing * col_count.saturating_sub(1) as u16);
-            // Divide the remaining space between each column equally
-            vec![Constraint::Length(total_space / col_count.max(1) as u16); col_count]
+            // Divide the space between each column equally
+            vec![Constraint::Length(max_width / col_count.max(1) as u16); col_count]
```

Comment:
Q: if we made the spacers take away space before the relative constraints (ratio/percentage) are applied (or if this is the way things work already), then could this be expressed as `[Ratio(1, col_count); col_count]`?

Reply:
Right now `[Ratio(1, col_count); col_count]` does the same thing as `[Length(max_width / col_count); col_count]`, but that may not be true in the future. In #788, `Length` is higher priority than `Ratio`, so I'd prefer we use `Length`.

> spacers take away space before the relative constraints

I'm not sure if I'm not following what you asking here specifically. I think it won't be possible to make spacers be higher priority because then spacers would allocate the entire space over ratio or percentage.

Reply:
What I mean is that if you have [Percentage(50); 2] and spacer = 10 splitting a space of 100, then you really don't have a 50% constraint, the constraints are 45% now. And in fact you can likely never write a percentage that meaningfully makes sense when spacers are taking some of that space. So that implies that the spacers should remove the space available to split before the constraints get applied. The split impl would then need to first remove the spacer space leaving 90 to split up, and then when that calculation of percentage -> solver constraint happens it would be a percentage of that instead of the total area.
```rust
                    let percent = f64::from(p) / 100.00;
                    solver.add_constraint(element.size() | EQ(STRONG) | (area_size * percent))?;
```

I haven't looked at the implementation to see if that's how this currently works though, or just how I'd intuitively think about it naively (which is perhaps the right way to think of this for our users).

Reply:
Right now isn't that just because of how the constraints interact and then fail when they can't all be met though?
That [p(100), min(20) is 3 cassowary constraints: [a=100(strong), b>=20(strong), b=20(medium)], which just happens to end up with [80,20]. Same deal with spacers I think right [a = 100(strong), sp = 1(required-1),..]

Reply:
> isn't that just because of how the constraints interact

Yes. But also, I don't see how it can work well any other way.

As context, I don't think it is possible to make `Percentage` (or `Length` or `Ratio`) higher priority than `Min` / `Max` because if you do you'll end up in cases where `Min` and `Max` stop working as intended. 

At best, `Length`, `Percentage` and `Ratio` can be the same priority as `Min` and `Max` (which is what we have now). In #788 `Length` is the same priority as `Min` and `Max` but `Percentage` and `Ratio` are lower priority. 

Because `Percentage` can't have a higher priority, and because it can have a lower priority, it is easier to think of `Percentage(x)` as filling x % of available space after higher priority constraints are resolved.

Reply:
Got it - resolving this for now - it was mostly curiosity driven.